### PR TITLE
Load aliases for Psych 4 (included in Ruby 3.1) 

### PR DESCRIPTION
**Pull request by ohbarye**
## Issue

N/A

## Description of changes

`YAML.load` is incompatible between Psych v4 and priors because Psych v4 has a breaking change.

ref: https://bugs.ruby-lang.org/issues/17866

To keep the current gem's behavior compatible between Psych v4 and priors (Ruby 3.1.0 and priors), it's necessary to use `YAML.unsafe_load` for Psych v4.

## Testing

All tests passed on my local with Ruby 3.1.2.
But I found that this repository's CI does not run tests on Ruby 3.1, so I'll try to add the version to [ci.yml](https://github.com/aws/aws-sdk-rails/blob/11c86231e2466d71854000d13162222fb50bec24/.github/workflows/ci.yml#L18).

---

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license.

### CODE REVIEW

File: test/aws/rails/sqs_active_job/configuration_test.rb [hunk 1]
```diff
@@ -36,6 +36,16 @@ module SqsActiveJob
...
         end
 
+        it 'accepts YAML config with alias' do
+          allow_any_instance_of(ERB).to receive(:result).and_return(<<~YAML)
+common: &common
+  default: 'https://queue-url'
+queues:
+  <<: *common
+          YAML
+          expect { Aws::Rails::SqsActiveJob::Configuration.new }.to_not raise_error
```
Comment (ohbarye) [inline test/aws/rails/sqs_active_job/configuration_test.rb:46]: If `YAML.load` for such a YAML runs on Ruby 3.1.0, it raises `Psych::BadAlias`. This is the breaking change. (Not only for alias though)

```
Failure:
Aws::Rails::SqsActiveJob::Configuration#test_0005_merges runtime options with YAML options [/Users/m-ohba/ghq/github.com/ohbarye/aws-sdk-rails/gemfiles/vendor/bundle/ruby/3.1.0/gems/rspec-expectations-3.12.0/lib/rspec/expectations/handler.rb:40]:
expected no Exception, got #<Psych::BadAlias: Unknown alias: common> with backtrace:
  # /Users/m-ohba/.rbenv/versions/3.1.2/lib/ruby/3.1.0/psych/visitors/to_ruby.rb:430:in `visit_Psych_Nodes_Alias'
  # /Users/m-ohba/.rbenv/versions/3.1.2/lib/ruby/3.1.0/psych/visitors/visitor.rb:30:in `visit'
  # /Users/m-ohba/.rbenv/versions/3.1.2/lib/ruby/3.1.0/psych/visitors/visitor.rb:6:in `accept'
  # /Users/m-ohba/.rbenv/versions/3.1.2/lib/ruby/3.1.0/psych/visitors/to_ruby.rb:35:in `accept'
  # /Users/m-ohba/.rbenv/versions/3.1.2/lib/ruby/3.1.0/psych/visitors/to_ruby.rb:345:in `block in revive_hash'
  # /Users/m-ohba/.rbenv/versions/3.1.2/lib/ruby/3.1.0/psych/visitors/to_ruby.rb:343:in `each'
  # /Users/m-ohba/.rbenv/versions/3.1.2/lib/ruby/3.1.0/psych/visitors/to_ruby.rb:343:in `each_slice'
  # /Users/m-ohba/.rbenv/versions/3.1.2/lib/ruby/3.1.0/psych/visitors/to_ruby.rb:343:in `revive_hash'
  # /Users/m-ohba/.rbenv/versions/3.1.2/lib/ruby/3.1.0/psych/visitors/to_ruby.rb:167:in `visit_Psych_Nodes_Mapping'
  # /Users/m-ohba/.rbenv/versions/3.1.2/lib/ruby/3.1.0/psych/visitors/visitor.rb:30:in `visit'
  # /Users/m-ohba/.rbenv/versions/3.1.2/lib/ruby/3.1.0/psych/visitors/visitor.rb:6:in `accept'
  # /Users/m-ohba/.rbenv/versions/3.1.2/lib/ruby/3.1.0/psych/visitors/to_ruby.rb:35:in `accept'
  # /Users/m-ohba/.rbenv/versions/3.1.2/lib/ruby/3.1.0/psych/visitors/to_ruby.rb:345:in `block in revive_hash'
  # /Users/m-ohba/.rbenv/versions/3.1.2/lib/ruby/3.1.0/psych/visitors/to_ruby.rb:343:in `each'
  # /Users/m-ohba/.rbenv/versions/3.1.2/lib/ruby/3.1.0/psych/visitors/to_ruby.rb:343:in `each_slice'
  # /Users/m-ohba/.rbenv/versions/3.1.2/lib/ruby/3.1.0/psych/visitors/to_ruby.rb:343:in `revive_hash'
  # /Users/m-ohba/.rbenv/versions/3.1.2/lib/ruby/3.1.0/psych/visitors/to_ruby.rb:167:in `visit_Psych_Nodes_Mapping'
  # /Users/m-ohba/.rbenv/versions/3.1.2/lib/ruby/3.1.0/psych/visitors/visitor.rb:30:in `visit'
  # /Users/m-ohba/.rbenv/versions/3.1.2/lib/ruby/3.1.0/psych/visitors/visitor.rb:6:in `accept'
  # /Users/m-ohba/.rbenv/versions/3.1.2/lib/ruby/3.1.0/psych/visitors/to_ruby.rb:35:in `accept'
  # /Users/m-ohba/.rbenv/versions/3.1.2/lib/ruby/3.1.0/psych/visitors/to_ruby.rb:318:in `visit_Psych_Nodes_Document'
  # /Users/m-ohba/.rbenv/versions/3.1.2/lib/ruby/3.1.0/psych/visitors/visitor.rb:30:in `visit'
  # /Users/m-ohba/.rbenv/versions/3.1.2/lib/ruby/3.1.0/psych/visitors/visitor.rb:6:in `accept'
  # /Users/m-ohba/.rbenv/versions/3.1.2/lib/ruby/3.1.0/psych/visitors/to_ruby.rb:35:in `accept'
  # /Users/m-ohba/.rbenv/versions/3.1.2/lib/ruby/3.1.0/psych.rb:335:in `safe_load'
  # /Users/m-ohba/.rbenv/versions/3.1.2/lib/ruby/3.1.0/psych.rb:370:in `load'
  # /Users/m-ohba/ghq/github.com/ohbarye/aws-sdk-rails/lib/aws/rails/sqs_active_job/configuration.rb:171:in `load_yaml'
```

File: lib/aws/rails/sqs_active_job/configuration.rb [hunk 2]
```diff
@@ -158,6 +157,18 @@ def config_file_path(options)
...
           "ft/aws-sdk-rails-activejob/#{Aws::Rails::VERSION}"
         end
+
+        def load_yaml(file_path)
+          require "erb"
+
+          # Avoid incompatible changes with Psych 4.0.0
+          # https://bugs.ruby-lang.org/issues/17866
+          if YAML.respond_to?(:unsafe_load)
+            YAML.unsafe_load(ERB.new(File.read(file_path)).result)
```
Comment (alextwoods) [inline lib/aws/rails/sqs_active_job/configuration.rb:167]: As this code was originally take from rails, I'd prefer to take their fix for the issue and use `YAML.load(..., aliases: true)` as in https://github.com/rails/rails/pull/42249

Comment (mullermp) [reply lib/aws/rails/sqs_active_job/configuration.rb:167]: Agree... Let's not use unsafe loading!

Comment (ohbarye) [reply lib/aws/rails/sqs_active_job/configuration.rb]: @alextwoods @mullermp 
Got it! I've changed code to use the Rails' approach you suggested. [19fafe7](https://github.com/aws/aws-sdk-rails/pull/91/commits/19fafe76993824b6c67598b00706f6938ec888a5)
Thank you for pointing it out.

File: test/aws/rails/sqs_active_job/configuration_test.rb (hunk 1 above)
Comment (mullermp) [reply test/aws/rails/sqs_active_job/configuration_test.rb:46]: This makes sense. Could we add a comment in code to this test definition that briefly references this issue OR explains why we need to test it?

**Comment by mullermp** [comment]
Thanks for the contribution and fixes!

**Comment by ohbarye** [comment]
Thanks for checking the CI & your review 😄

**Pull request merged**